### PR TITLE
add request/response bytes report.

### PR DESCRIPTION
**Pull request by rnewson**
Ben,

I added two reports that I find helpful. I had to remove the headerLines thing as it doesn't seem to work in latest node anyway and I needed the original content-length header.

I'm also looking for a neat way to tolerate errors (you can catch them with client.on('error', function(err)) but the client should then be replaced with a new way). Interested in your suggestions.

### CODE REVIEW

**Comment by jonjlee** [comment]
Robert,

Thanks for the updates! The bytes sent/received report looks great and will be useful. I need to take a peek at the request headers change though, because I want to verify that we're capturing all the headers that node sends in the case node adds headers beyond what the user specifies.

I'll get back to you on the client error handling. I haven't spent time poking at that yet, and I think that the node behavior for errors changed recently. In the meantime, let me know if you have ideas for a good solution.

I'll pull in the changes later today.

Thanks,
Jonathan

**Comment by rnewson** [comment]
You're very welcome. I'm finding nodeloadlib.js a very useful toolkit.

I'm working on a few things with it but I've noticed that sometimes it doesn't honor the timeLimit (it just keeps going).

It would be neat to capture all the headers but I didn't think node added any of its own (perhaps it used to?)

**Comment by rnewson** [comment]
so checkConditions returns false (for the timeLimit function) but the requests keep going. This is against one particular http server (outside of my control) but not against another. So some server-side behavior is preventing the test from terminating, maybe? Will start another pull request if I find a fix.

**Comment by jonjlee** [comment]
Glad you're finding the library useful! We can always use more help working on it :).

node does in fact add it's own headers. Inside node/lib/http.js, OutgoingMessage.prototype._storeHeader may add Connection and Transfer-Encoding headers. I've found the precise request header info to be invaluable when debugging load tests, so I'm looking into how we can extract it.

I've definitely seen the timeLimit problem happen before as well.  I'll open an issue to track it.

**Comment by rnewson** [comment]
Commenting here for context.

I add both of those headers myself, which is why I guess I've not noticed that behavior.

I couldn't quite follow the addJob/clone area but it seems only one of the concurrent jobs is a ConditionalLoop. I think that's deliberate if a little hard to follow. Anyway, I observe that the callback doesn't fire, so job.done isn't set to true, so the scheduler doesn't shut down the job group. If that helps at all.
